src/Microsoft.PowerShell.ConsoleHost/host/msh/CommandLineParameterParser.cs

```diff
@@ -548,6 +548,15 @@ private void ParseHelper(string[] args)
                     _noInteractive = true;
                     _skipUserInit = true;
                     _noExit = false;
+
+                    ++i;
+                    if (i < args.Length)
+                    {
+                        if (LanguagePrimitives.TryConvertTo<int>(args[i], out int verNumber) && (verNumber == 2 || verNumber == 3))
```

Comment:
I think we should error on any value passed to `-v` as part of this PR.

Reply:
Done.